### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by wolfpilot**
This PR will:

- Add the npm package lock file to ensure that dependencies are consistent across all machines and platforms.
- Remove unused `gulp-tap` dependency.
- Update all other dependencies to their latest versions (exception noted below).

A few notes:
1. `gulp-transform` now requires character encoding in order to render the docs properly. I tried **utf8** and as far as I can tell it's working fine.
2. `stylelint-config-standard` introduces two new rules.
   - `at-rule-no-unknown`, by default, throws errors when linting SCSS directives (@if, @for, @each, etc.). I've set it to exclude linting via the CSS linter, but use the SCSS one instead.
   - `no-descending-specificity` a.k.a. enforcing good habits regarding CSS specificity. More info [here](https://stylelint.io/user-guide/rules/no-descending-specificity/). Leaving this feature enabled by default is something that we _may not_ want.

Please also note that I have not yet updated `uglify-js` since v3 **is not backwards compatible** with v2 and I couldn't wrap my head around the new API. It would be great if anyone's willing to help with this, perhaps in a new PR.

Tested every Gulp task on a real project of roughly 50 components and everything's working smoothly.

### CODE REVIEW

**Comment by wolfpilot** [comment]
@wouterkroes I updated gulp to v3.9.0 and the warning's gone now.

Babel changed some stuff making `babel-core/register` a required dependency. Normally, it should have been them who should have added it as a dependency, but since Gulp also relies on Babel and the Babel team didn't do jack about it, Gulp added it as a dependency to their own project.

HOWEVER, they still kept getting new issues opened by people who neither updated their Gulp version, nor added `babel-core/register` as a separate dependency, so the Gulp devs got fed up and eventually added that "error" log asking people to install `babel-core/register` - [link](https://github.com/gulpjs/gulp/issues/1631). So... now, it's very much hard-coded in red starting Gulp v3.9.1, even though [it's only a warning](https://github.com/gulpjs/gulp-cli/issues/142). To quote: 

> It seems that some consumers are allergic to the color red - we should revisit the coloring and messaging of the requireFail log.

**TL;DR:** 3.9.0 works fine. 3.9.1 also works, but it shows a stupid hard-coded log.

**Comment by wolfpilot** [comment]
As a final touch to this PR,  the `npm scripts` have been updated to include all main Gulp tasks.

Calling `npm run dev` instead of `gulp dev` will ensure that the user's local version of Gulp is being used and therefore avoid any undesired conflicts with external dependencies. Docs have also been updated to reflect this, as well as advising the global installation of `gulp-cli`, and **not** `gulp`.

List of scripts below:

`npm run dev`
`npm run dist`
`npm run lint`
`npm run test`
`npm run upload`

File: readme.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
     - [Node.js](http://nodejs.org/)
 2. Run this command to install the global dependencies:
```
Comment (branneman) [inline readme.md]: Actually, this whole step is not necessary anymore, now that we're not using the global gulp. Can you remove it?

Comment (wolfpilot) [reply readme.md]: True, I was still thinking how else would you call gulp without the CLI. Silly me :smile:
